corehq/apps/api/resources/messaging_event/utils.py

```diff
@@ -27,3 +36,33 @@ def sort_query(query, request_data):
         order_by_args = [order_by, "id"]
 
     return query.order_by(*order_by_args)
+
+
+class CursorParams(namedtuple("CursorParams", "params, domain, is_cursor")):
+    def __getitem__(self, key):
+        return self.params[key]
+
+    def __contains__(self, key):
```

Comment:
These methods typically have a different meaning on `namedtuple` objects. What do you think of using `attrs` rather than `namedtuple` for this class to avoid conflating dict-like access patterns on a tuple-based thing?

Reply:
48ab71a5ee7

Reply:
It was briefly confusing that that commit is not in this PR. It appears to have made it into https://github.com/dimagi/commcare-hq/pull/29885 though, so I guess all is well.